Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Returning forces for all atoms - regardless of whether "fixed" or "free"
i_structure.get_forces(apply_constraint=False)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mPropertyNotImplementedError[0m               Traceback (most recent call last)
Cell [0;32mIn[18], line 2[0m
[1;32m      1[0m [38;5;66;03m# Returning forces for all atoms - regardless of whether "fixed" or "free"[39;00m
[0;32m----> 2[0m [43mi_structure[49m[38;5;241;43m.[39;49m[43mget_forces[49m[43m([49m[43mapply_constraint[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ase/atoms.py:812[0m, in [0;36mAtoms.get_forces[0;34m(self, apply_constraint, md)[0m
[1;32m    810[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_calc [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    811[0m     [38;5;28;01mraise[39;00m [38;5;167;01mRuntimeError[39;00m([38;5;124m'[39m[38;5;124mAtoms object has no calculator.[39m[38;5;124m'[39m)
[0;32m--> 812[0m forces [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_calc[49m[38;5;241;43m.[39;49m[43mget_forces[49m[43m([49m[38;5;28;43mself[39;49m[43m)[49m
[1;32m    814[0m [38;5;28;01mif[39;00m apply_constraint:
[1;32m    815[0m     [38;5;66;03m# We need a special md flag here because for MD we want[39;00m
[1;32m    816[0m     [38;5;66;03m# to skip real constraints but include special "constraints"[39;00m
[1;32m    817[0m     [38;5;66;03m# Like Hookean.[39;00m
[1;32m    818[0m     [38;5;28;01mfor[39;00m constraint [38;5;129;01min[39;00m [38;5;28mself[39m[38;5;241m.[39mconstraints:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ase/calculators/abc.py:30[0m, in [0;36mGetPropertiesMixin.get_forces[0;34m(self, atoms)[0m
[1;32m     29[0m [38;5;28;01mdef[39;00m [38;5;21mget_forces[39m([38;5;28mself[39m, atoms[38;5;241m=[39m[38;5;28;01mNone[39;00m):
[0;32m---> 30[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mget_property[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mforces[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43matoms[49m[43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ase/calculators/singlepoint.py:49[0m, in [0;36mSinglePointCalculator.get_property[0;34m(self, name, atoms, allow_calculation)[0m
[1;32m     47[0m [38;5;28;01mif[39;00m name [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m [38;5;28mself[39m[38;5;241m.[39mresults [38;5;129;01mor[39;00m [38;5;28mself[39m[38;5;241m.[39mcheck_state(atoms):
[1;32m     48[0m     [38;5;28;01mif[39;00m allow_calculation:
[0;32m---> 49[0m         [38;5;28;01mraise[39;00m PropertyNotImplementedError(
[1;32m     50[0m             [38;5;124mf[39m[38;5;124m'[39m[38;5;124mThe property [39m[38;5;124m"[39m[38;5;132;01m{[39;00mname[38;5;132;01m}[39;00m[38;5;124m"[39m[38;5;124m is not available.[39m[38;5;124m'[39m)
[1;32m     51[0m     [38;5;28;01mreturn[39;00m [38;5;28;01mNone[39;00m
[1;32m     53[0m result [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mresults[name]

[0;31mPropertyNotImplementedError[0m: The property "forces" is not available.

